src/main/java/me/zeroeightsix/kami/gui/kami/theme/kami/KamiFrameUI.java

```diff
@@ -38,18 +41,54 @@
     boolean centerX = false;
     boolean centerY = false;
     int xLineOffset = 0;
-
+    Color rgbGuiColor = new Color(255,0,0);
+    private boolean rgbListGenerated = false;
+    public boolean rgbGuiEnabled = false;
+    private Color[] rgbColors;
+    private int rgbMax = 0;
+    private int cycleNum = 0;
+    private int cycleCycleNum = 0;
+    /**
+     *  speed value between 0 and 100
+     */
+    public int speed = 0;
     private static final RootFontRenderer ff = new RootLargeFontRenderer();
-
     @Override
     public void renderComponent(Frame component, FontRenderer fontRenderer) {
+        //generate RGB list
+        if (!rgbListGenerated)
+        {
+            List<Color> colorsList = new ArrayList<Color>();
+            for (int r=0; r<100; r++) colorsList.add(new Color(r * 255 / 100, 255, 0));
+            for (int g=100; g>0; g--) colorsList.add(new Color(255, g*255/100,0));
+            for (int b=0; b<100; b++) colorsList.add(new Color(255, 0, b * 255 / 100));
+            for (int r=100; r>0; r--) colorsList.add(new Color(r*255/100,0,255));
+            for (int g=0; g<100; g++) colorsList.add(new Color(0,g*255/100,255));
+            for (int b=100; b>0; b--) colorsList.add(new Color(0,255,b*255/100));
+            colorsList.add(new Color(0,255,0));
+            rgbColors = colorsList.toArray(new Color[colorsList.size()]);
+            rgbListGenerated = true;
+        }
```

Comment:
Oh my fucking god WHY
Please, just use the RGB code from ActiveModules, it's legit one line.